xbmc/storage/IStorageProvider.h

```diff
@@ -21,7 +21,12 @@ class IStorageEventsCallback
 public:
   virtual ~IStorageEventsCallback() = default;
 
-  virtual void OnStorageAdded(const std::string &label, const std::string &path) = 0;
+  /*! \brief Callback executed when a new storage device is added
+    * @param label the storage label
+    * @param path the storage path
+    * @param optical if this is an optical device
+    */
+  virtual void OnStorageAdded(const std::string& label, const std::string& path, bool optical) = 0;
```

Comment:
or even better go with `enum class Optical { YES, NO }` instead of a new bool argument